package/libsolv.spec.in

```diff
@@ -190,83 +165,80 @@ Group:          Development/Languages/Python
 Python3 bindings for libsolv.
 
 %package -n perl-solv
-Requires:       perl = %{perl_version}
 Summary:        Perl bindings for the libsolv library
 Group:          Development/Languages/Perl
+Requires:       perl = %{perl_version}
 
 %description -n perl-solv
 Perl bindings for libsolv.
 
 %prep
-%setup -n libsolv-%{version}
+%setup -q
 
 %build
 export CFLAGS="%{optflags}"
 export CXXFLAGS="$CFLAGS"
 
 CMAKE_FLAGS=
-%if 0%{?fedora} || 0%{?rhel} >= 6
-CMAKE_FLAGS="-DFEDORA=1 -DENABLE_APPDATA=1 -DENABLE_COMPS=1"
+%if 0%{?fedora} || 0%{?rhel}
+CMAKE_FLAGS="-DFEDORA=1"
 %endif
 %if 0%{?mageia}
-CMAKE_FLAGS="-DMAGEIA=1 -DENABLE_APPDATA=1 -DENABLE_COMPS=1"
+CMAKE_FLAGS="-DMAGEIA=1"
 %endif
 %if 0%{?suse_version}
-CMAKE_FLAGS="-DSUSE=1 -DENABLE_APPDATA=1 -DENABLE_COMPS=1"
+CMAKE_FLAGS="-DSUSE=1"
 %endif
 
-cmake	$CMAKE_FLAGS \
-	-DCMAKE_INSTALL_PREFIX=%{_prefix} \
-	-DLIB=%{_lib} \
-	-DCMAKE_VERBOSE_MAKEFILE=TRUE \
-	-DCMAKE_BUILD_TYPE=RelWithDebInfo \
+# SHARE/MODULE flags in SUSE enforce no-undefined which is not applicable for
+# bindings here
+%cmake $CMAKE_FLAGS \
+	-DCMAKE_MODULE_LINKER_FLAGS="-Wl,--as-needed -Wl,-z,now" \
+	-DCMAKE_SHARED_LINKER_FLAGS="-Wl,--as-needed -Wl,-z,now" \
```

Comment:
The `%cmake` macro behaves differently between RH/Fedora and SUSE. Namely, SUSE's version enforces out of tree builds and Fedora's doesn't. That's why it's not used here.

As for the MODULE/SHARED linker flags, just export them up where you export CFLAGS. CMake will read them there just fine.

Reply:
LINKER_FLAGS are not tied to the CFLAGS at all. They need to be defined on their own.